ompi/runtime/ompi_rte.c

```diff
@@ -78,7 +78,7 @@ static int _setup_proc_session_dir(char **sdir);
 #define OPAL_PRINT_NAME_ARG_NUM_BUFS    16
 
 static bool fns_init=false;
-static opal_tsd_key_t print_args_tsd_key;
+static opal_tsd_tracked_key_t *print_args_tsd_key = NULL;
```

Comment:
Any particular reason to transform this into a pointer ? You could have used OBJ_CONSTRUCT/DESTRUCT instead of NEW_RELEASE as you did for others.

Reply:
Object is not always constructed. So to make checking easier we opted to use pointers. We did the same in opal/util/net.c. In wpool the objects is always constructed so there is no need to check if it was initialized.

Reply:
Checking for the pointer being NULL or checking for the value of fns_init would have been equivalent. Anyway, either way is fine, I was just curious for the reason.

Reply:
Oh duh, yeah you’re right - I think for consistency it may be better to revert it. Thanks!